integration-test/src/test/java/com/lohika/jclub/integration/BaseIntegrationTest.java

```
@@ -0,0 +1,87 @@
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+
+import javax.annotation.PostConstruct;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@Log4j
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = IntegrationTestApplication.class)
+@ContextConfiguration(initializers = BaseIntegrationTest.Initializer.class)
+public abstract class BaseIntegrationTest {
+
+  protected static final String DSL_EXECUTOR_SERVICE = "DSL-EXECUTOR-SERVICE";
+  protected static final String API_GATEWAY_SERVICE = "API-GATEWAY-SERVICE";
+
+  private static final String DISCOVERY_SERVER = "localhost";
+  private static final int DISCOVERY_SERVER_PORT = 8761;
+  private static final int SLEEP = 5000;
+
+  @Autowired
+  protected StorageServiceClient storageServiceClient;
+
+  @BeforeClass
+  public static void init() {
+    assertThat(EntToEndIntegrationTestSuite.environment, notNullValue());
+  }
+
+  @Before
+  public void warmUp() {
+    storageServiceClient.list();
+  }
+
+  @Before
+  public void cleanUp() {
+    storageServiceClient.list().getContent()
+        .forEach(apartment -> storageServiceClient.delete(apartment.getId()));
+  }
+
+  public static void waitFor(String service) throws InterruptedException {
+    HttpClient client = HttpClientBuilder.create().build();
+    String gateway = "http://" + DISCOVERY_SERVER + ":" + DISCOVERY_SERVER_PORT + "/eureka/apps/" + service;
+    HttpGet httpGet = new HttpGet(gateway);
+
+    log.info("Wait for :" + service);
+    HttpResponse httpResponse = null;
+    do {
+      Thread.sleep(SLEEP);
+      try {
+        httpResponse = client.execute(httpGet);
+      } catch (IOException ignored) {
```

Maybe throw it away instead of ignoring?